internal/fork/ietf-cms/verify.go

```diff
@@ -120,6 +122,23 @@ func (sd *SignedData) verify(econtent []byte, opts x509.VerifyOptions, tsOpts x5
 			return nil, err
 		}
 
+		// Handle certificates where the Subject Alternative Name is not set to
+		// a supported GeneralName (RFC 5280 4.2.1.6). Go only supports DNS, IP
+		// addresses, email addresses, or URIs as SANs. Fulcio can issue a
+		// certificate with an OtherName GeneralName, so remove the unhandled
+		// critical SAN extension before verifying.
+		// This matches https://github.com/sigstore/cosign/blob/a0752eb40b500316ac417baf4926a2c2d99b39b8/pkg/cosign/verify.go#L236-L248
+		if len(cert.UnhandledCriticalExtensions) > 0 {
+			var unhandledExts []asn1.ObjectIdentifier
+			for _, oid := range cert.UnhandledCriticalExtensions {
+				if !oid.Equal(cryptoutils.SANOID) {
+					unhandledExts = append(unhandledExts, oid)
+				}
+			}
+
+			cert.UnhandledCriticalExtensions = unhandledExts
```

Comment:
This is filtering out the SANOID, but should we be handling it elsewhere?

Reply:
We're only filtering it out from the list of unhandled extensions. The value is still matched from cert.Extensions. Not rigorous but `gitsign verify <>` correctly handles when given the right identity and errors out when given the wrong identity where before it was complaining about the unhandled extension. I think there's some mismatch in that a parser somewhere is able to load the value into cert.Extensions but somewhere else (the go std lib?) it's marked as an unhandled ext?

Reply:
https://pkg.go.dev/crypto/x509#Certificate
```
	// Users can access these extensions using Extensions and can remove
	// elements from this slice if they believe that they have been
	// handled.
```

👍 LGTM